pkg/controller/proxyconfig/validation.go

```diff
@@ -0,0 +1,52 @@
+package proxyconfig
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-network-operator/pkg/util/validation"
+)
+
+const (
+	proxyHTTPScheme = "http"
+	// noProxyWildcard is the string used to as a wildcard attached to a
+	// domain suffix in proxy.spec.noProxy to bypass proxying.
+	noProxyWildcard = "*"
+)
+
+// ValidateProxyConfig ensures that proxyConfig is valid.
+func (r *ReconcileProxyConfig) ValidateProxyConfig(proxyConfig *configv1.ProxySpec) error {
+	if isSpecHTTPProxySet(proxyConfig) {
+		scheme, err := validation.URI(proxyConfig.HTTPProxy)
+		if err != nil {
+			return fmt.Errorf("invalid httpProxy URI: %v", err)
+		}
+		if scheme != proxyHTTPScheme {
+			return fmt.Errorf("httpProxy requires an %q URI scheme", proxyHTTPScheme)
+		}
+	}
+
+	if isSpecHTTPSProxySet(proxyConfig) {
+		_, err := validation.URI(proxyConfig.HTTPSProxy)
```

Comment:
should still confirm the scheme is at least `http` or `https`?

Reply:
ack